jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/async/AsyncPool.java

```diff
@@ -63,9 +70,14 @@ public AsyncTask submit(Runnable task, String displayName, DataService dataServi
                 return null;
             };
             AsyncTask asyncTask = new AsyncTask(c, this, taskId, displayName, dataService, requestId);
-            /* Future<Object> future = */ executor.submit(asyncTask);
-            runningTasks.put(taskId, asyncTask);
-            return asyncTask;
+            try {
+                /* Future<Object> future = */ executor.submit(asyncTask);
+                runningTasks.put(taskId, asyncTask);
+                return asyncTask;
+            } catch (RejectedExecutionException ex) {
+                ServletOps.errorBadRequest("Async task request rejected");
```

Comment:
Should this be more specific so users understand why it was rejected e.g. add something like ` - already <maxTasks> in progress`

Reply:
OK